src/nvim/api/private/helpers.c

```diff
@@ -971,3 +970,19 @@ static void set_option_value_err(char *key,
     api_set_error(err, Exception, "%s", errmsg);
   }
 }
+
+void _api_set_error(Error *err, ErrorType errType, const char *format, ...)
+{
+    va_list args1;
+    va_start(args1, format);
+    va_list args2;
+    va_copy(args2, args1);
+    int len = vsnprintf(NULL, 0, format, args1);
+    va_end(args1);
+	err->msg = xmalloc(len+1);
+    vsprintf(err->msg, format, args2);
+    va_end(args2);
+	
+	err->set = true;
```

Comment:
Are you sure you can unconditionally set this to `true`? I see only one such instance where you actually removed `err->set = true` (channel.c, line 222); what about the others?